iOS_SDK/OneSignalSDK/Source/OSMessagingController.m

```diff
@@ -531,11 +533,35 @@ - (void)messageViewDidSelectAction:(OSInAppMessage *)message withAction:(OSInApp
     if (self.actionClickBlock)
         self.actionClickBlock(action);
     
+    if (message.isPreview) {
+        [self processPreviewInAppMessage:message withAction:action];
+        return;
+    }
+
+    // The following features are for non preview IAM
+    // Make sure no click, outcome, tag tracking is performed for IAM previews
     [self sendClickRESTCall:message withAction:action];
     [self sendTagCallWithAction:action];
     [self sendOutcomes:action.outcomes];
 }
 
+/*
+* Show the developer what will happen with a non IAM preview
+ */
+- (void)processPreviewInAppMessage:(OSInAppMessage *)message withAction:(OSInAppMessageAction *)action {
+     if (action.tags) {
+        OSInAppMessageTag *tag = action.tags;
+        if (tag.tagsToAdd)
```

Comment:
Log all tags at once, no need to differentiate or add extra code. Lets do `tag.description` or `tag.jsonRepresentation` so we can just print a string of tags and they can look at the payload.

I have a similar comment on Android PR for this work